src/components/Loader.vue

```diff
@@ -0,0 +1,315 @@
+                    <p v-if="message" class="message nq-text">{{ message }}</p>
+                </div>
+
+                <div class="action-row">
+                    <button v-if="mainAction" class="main-action nq-button red inverse" @click="onMainAction">{{ mainAction }}</button>
+                    <a v-if="alternativeAction" href="javascript:void(0)" class="alternative-action nq-link" @click="onAlternativeAction">{{ alternativeAction }}</a>
+                </div>
+            </div>
+        </transition>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Watch, Emit, Vue } from 'vue-property-decorator';
+
+@Component
+export default class Loader extends Vue {
+    private static LOADING = 'loading';
+    private static SUCCESS = 'success';
+    private static WARNING = 'warning';
+    private static ERROR   = 'error';
+
+    @Prop({type: String, default: 'Improving the world'}) private title!: string;
+    @Prop({type: String, default: Loader.LOADING}) private state!: string;
+    @Prop(Boolean) private lightBlue?: boolean;
+    @Prop(String) private status?: string;
+    @Prop(String) private message?: string;
+    @Prop(String) private mainAction?: string;
+    @Prop(String) private alternativeAction?: string;
+
+    private currentStatus: string = '';
+    private nextStatus: string = '';
+    private isStatusTransitioning: boolean = false;
+
+    /**
+     * To enable a smooth transition of the non-transitionable background-image
+     * property, we instead place the new background above the old one and
+     * animate the top element's opacity. But because the color area has rounded
+     * corners, and the browser creates transparent pixels in the corner
+     * because of anti-aliasing, the blue background partly shines through the
+     * transparent corner pixels of the foreground. Thus we remove the background
+     * color after the transition is complete.
+     */
+    private showLoadingBackground: boolean = true;
+
+    private _loadingTitle?: string;
+
+    private get loadingTitle() {
+        return this._loadingTitle || (this._loadingTitle = this.title);
+    }
+
+    @Watch('state', {immediate: true})
+    private updateState(newState: string, oldState: string) {
+        // When the component is initialized with a state other than LOADING
+        if (!oldState && (newState !== Loader.LOADING)) {
+            this.showLoadingBackground = false;
+        }
+
+        // When the state changes later and animates
+        if (oldState && (newState !== Loader.LOADING)) {
+            setTimeout(() => this.showLoadingBackground = false, 1000);
+        }
+    }
+
+    @Watch('status', {immediate: true})
+    private updateStatus(newStatus: string, oldStatus: string) {
+        this.nextStatus = newStatus;
+        this.isStatusTransitioning = true;
+
+        setTimeout(() => {
+            this.currentStatus = newStatus;
+            this.isStatusTransitioning = false;
+        }, 500);
```

Comment:
what happens if i Change status before that timeout.

Reply:
I don't know, let's find out! :laughing:

Reply:
It works. Looks not optimal, but works for now and displays the correct status in the correct order.